### PR TITLE
FAI-7473 Rework Asana converters

### PR DESCRIPTION
**Pull request by ypc-faros**
- Reworks most of the Asana converters with improvements and bug fixes.
- Removes the `Stories` converter, which was incorrectly associated with tasks. [Stories](https://developers.asana.com/reference/stories) represent an activity associated with an object in the Asana system. Maybe we can find a good use for them in the future.

### CODE REVIEW

**Review by omreego — APPROVED** [review]
Looks good

File: destinations/airbyte-faros-destination/src/converters/asana/common.ts [hunk 1]
```diff
@@ -35,17 +35,20 @@ export class AsanaCommon {
...
 
-  static toTmsTaskType(type: string): TmsTaskType {
-    const detail = type.toLowerCase();
-    switch (detail) {
+  static normalize(str: string): string {
+    return str.replace(/\s/g, '').toLowerCase();
+  }
+
+  static toTmsTaskType(resource_type: string): TmsTaskType {
+    switch (this.normalize(resource_type)) {
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/asana/common.ts:43]:
```suggestion
    if (!resource_type) {
        return { category: Custom, detail: 'undefined' }
    }
    switch (this.normalize(resource_type)) {
```

File: destinations/airbyte-faros-destination/src/converters/asana/stories.ts [hunk 2]
```diff
@@ -1,36 +0,0 @@
...
-
-  async convert(
-    record: AirbyteRecord
-  ): Promise<ReadonlyArray<DestinationRecord>> {
-    const res: DestinationRecord[] = [];
-    const source = this.streamName.source;
-    const story = record.record.data;
-
-    res.push({
-      model: 'tms_Task',
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/asana/stories.ts:18]: I thought that Story could be mapped into a tms_Task of type Story, no?

File: destinations/airbyte-faros-destination/src/converters/asana/tasks.ts [hunk 3]
```diff
@@ -160,21 +153,16 @@ export class Tasks extends AsanaConverter {
...
-    return res;
+  private getStatus(task: Dictionary<any>): TmsTaskStatus | null {
+    const status = this.findFieldByName(task.custom_fields, 'status');
+
+    if (typeof status === 'string') {
+      return this.toTmsTaskStatus(status);
+    } else if (task.completed) {
+      return {category: Tms_TaskStatusCategory.Done, detail: 'completed'};
+    } else {
+      return null;
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/asana/tasks.ts:164]: better `{category: Custom, detail: 'undefined' }`

File: destinations/airbyte-faros-destination/src/converters/asana/tasks.ts [hunk 4]
```diff
@@ -205,16 +193,15 @@ export class Tasks extends AsanaConverter {
   }
 
   private toTmsTaskStatus(status: string): TmsTaskStatus {
-    const detail = status.toLowerCase();
-    switch (detail) {
+    switch (AsanaCommon.normalize(status)) {
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/asana/tasks.ts:196]: same

**Review by tovbinm — APPROVED** [review]

File: destinations/airbyte-faros-destination/src/converters/asana/stories.ts (hunk 2 above)
Comment (tovbinm) [reply destinations/airbyte-faros-destination/src/converters/asana/stories.ts:18]: OK, looks like Stories are just comments https://developers.asana.com/reference/stories ok, can be skipped then.

**Pull request merged**